### DISCUSSION TITLE
Problems setting `allow_forking` through REST API

### DISCUSSION THREAD

**Discussion by brtrick** [community/community#39101] [Apps, API and Webhooks]
According to this GitHub [blog post](https://github.blog/changelog/2021-10-11-set-via-api-whether-a-repository-allows-forking/),

> You can now set whether a repository allows forking when creating or updating it using either the REST or GraphQL API.

There are, however, a few issues with setting `allow_forking`:

1. Using the REST API, `allow_forking` can only be set when updating, not when creating. (It still gets ignored when creating, and `allow_forking` is not one of the values returned in the created repo's information.) Updating the blog post to reflect the actual situation would be helpful; updating the create_repo functionality to set `allow_forking` would be even better.

2. Since `allow_forking` can now be set--at least when updating--it would be great if that value was included in the generic repo information returns.

3. Finally, trying to set `allow_forking` to `false` on a public repo produces the following incorrect error message:

    ```422 - This organization does not allow private repository forking```

    In my case, the repo that it was attempting to PATCH was set to `private: false` and `allow_forking: true`, and the organization settings **allowed** the forking of private repositories. The problem was not that the organization forbid private repository forking but that the repository was public so forking could not be disabled.

**Comment by i-am-Sirius-Black** [comment]
You're absolutely right, and I've run into the same issues when trying to set allow_forking via the REST API.

When creating a repo using the REST API, allow_forking seems to be ignored entirely. It's not listed in the response, and the setting doesn't carry through. The blog post makes it sound like this should work during creation, but in practice, it only seems to apply during an update. It would be helpful if GitHub either updates the blog post or enables this during creation as expected.

I also agree it would be useful if allow_forking was included in the full repository object returned by various endpoints (like /repos/:owner/:repo). Right now, you have to make a separate call to confirm it.

The 422 error message when trying to set allow_forking: false on a public repo is definitely misleading. The message suggests it's an issue with private repo forking policies, which doesn't make sense in that context. It seems like this error should say something like "Forking cannot be disabled on public repositories" instead, if that's the actual reason.

Hopefully, GitHub sees this and updates both the docs and error messages for clarity. Thanks for raising it.
